docs/source/usage_guides/distributed_inference.md

```diff
@@ -134,3 +139,85 @@ with distributed_state.split_between_processes(["a dog", "a cat", "a chicken"],
 
 On the first GPU, the prompts will be `["a dog", "a cat"]`, and on the second GPU it will be `["a chicken", "a chicken"]`.
 Make sure to drop the final sample, as it will be a duplicate of the previous one.
+
+## A more memory-efficient version (experimental)
```

Comment:
This is a vague title. Let's rename to something a bit more clear. 
E.g. "Memory-efficient pipeline parallelism (experimental)"